guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,446 @@
+
+<!-- markdownlint-disable MD033 -->
+<script setup>
+import Callout from '../.vitepress/components/callout.vue'
+</script>
+
+:::tip
+<Callout />
+:::
+
+<!-- markdownlint-enable MD033 -->
+
+## 📝 In short, deploy a CosmWasm rollup
+
+This is a gist of the [CosmWasm rollup](../tutorials/cosmwasm.md) guide.
+
+```bash
+git clone --branch v0.50.0 --depth 1 https://github.com/CosmWasm/wasmd.git
+cd wasmd
+go mod edit -replace github.com/cosmos/cosmos-sdk=github.com/rollkit/cosmos-sdk@v0.50.6-rollkit-v0.13.3-no-fraud-proofs
+go mod tidy -compat=1.17
+go mod download
+```
+
+Comment out lines 898-900 in `app/app.go`:
+
+```bash
+echo 'diff --git a/app/app.go b/app/app.go
+index 5c67ba31..797b6bf4 100644
+--- a/app/app.go
++++ b/app/app.go
+@@ -895,9 +895,6 @@ func NewWasmApp(
+ 	// At startup, after all modules have been registered, check that all proto
+ 	// annotations are correct.
+ 	protoFiles, err := proto.MergedRegistry()
+-	if err != nil {
+-		panic(err)
+-	}
+ 	err = msgservice.ValidateProtoAnnotations(protoFiles)
+ 	if err != nil {
+ 		// Once we switch to using protoreflect-based antehandlers, we might' | git apply
+```
+
+Build `wasmd`:
+
+```bash
+make install
+```
+
+To start a local DA, run:
+
+```bash
+curl -sSL https://rollkit.dev/install-local-da.sh | bash -s v0.2.0
+```
+
+From inside the `wasmd` directory, run the init script:
+
+```bash
+curl -sSL https://rollkit.dev/cosmwasm/init.sh | perl -pe 's/127\.0\.0\.1/0.0.0.0/g' | sh
+```
+
+:::info
+We modify the listening address from `127.0.0.1` to `0.0.0.0` so that it is accessible from the Hyperlane docker agents later on.
+:::
+
+With that, we have kickstarted our `wasmd` network!
+
+## 💻 Deploy the Hyperlane contracts {#deploy-contracts}
+
+Fork the cw-hyperlane repo:
+
+```bash
+git clone --depth 1 git@github.com:many-things/cw-hyperlane.git
+cd cw-hyperlane
+```
+
+#### Create `config.yaml` with our networks setup:
+
+```bash
+echo 'networks:
+  - id: "localwasmd"
+    hrp: "wasm"
+    endpoint:
+      rpc: "http://localhost:36657"
+      rest: "http://localhost:1317"
+      grpc: "http://172.17.0.1:9290"
+    gas:
+      price: "0.025"
+      denom: "uwasm"
+    domain: 1063
+  - id: "stride-internal-1"
+    hrp: "stride"
+    endpoint:
+      rpc: "https://stride.testnet-1.stridenet.co"
+      rest: "https://stride.testnet-1.stridenet.co/api"
+      grpc: "http://stride-testnet-grpc.polkachu.com:12290"
+    gas:
+      price: "0"
+      denom: "ustrd"
+    domain: 1651
+
+# wasm133xh839fjn9wxzg6vhc0370lcem8939zr8uu45
+# stride133xh839fjn9wxzg6vhc0370lcem8939z2sd4gn
+signer: "join always addict position jungle jeans bus govern crack huge photo purse famous live velvet virtual weekend hire cricket media dignity wait load mercy"
+
+deploy:
```

Comment:
I'd add a clarifying comment that this is for the hyperlane deployment on localwasm